docs/howto/configure/auth-management.md

```diff
@@ -193,6 +193,77 @@ If administrators report a `403 forbidden` error when they try to log in to the
 In this case, they should go to the configuration page for this app within their GitHub organization and explicitly grant it access.
 See [the GitHub apps for organizations docs](https://docs.github.com/en/organizations/managing-access-to-your-organizations-apps) for more information.
 
+### Restricting user profiles based on GitHub Team Membership
+
+JupyterHub has support for using [profileList](https://zero-to-jupyterhub.readthedocs.io/en/latest/jupyterhub/customizing/user-environment.html#using-multiple-profiles-to-let-users-select-their-environment)
+to give users a choice of machine sizes and images to choose from when launching their
+server.
+
+In addition, we can allow people access to specific profiles based on their GitHub Teams membership!
+This only works if the hub is already set to allow people only from certain GitHub organizations
+to log in.
+
+The key `allowed_teams` can be set for any profile definition, with a list of GitHub
+teams (formatted as `<github-org>:<team-name>`) that will get access to that profile. Users
+need to be a member of any one of the listed teams for access. The list of teams a user
+is part of is fetched at login time - so if the user is added to a GitHub team, they need
+to log out and log back in to the JupyterHub (not necessarily to GitHub!) to see the new
+profiles they have access to. To remove access to a profile from a user, they have to be
```

Comment:
We should document this in docs.2i2c.org so that others know this is possible. I think we should also flag this as a `BETA` feature that may change at any moment (e.g. if we want to generalize this to non-github, or if we want to change the behavior around things like the need to delete users)

Reply:
@choldgraf opened https://github.com/2i2c-org/infrastructure/issues/1245 to track.